pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ConsumerHandler.java

```diff
@@ -467,8 +470,21 @@ protected ConsumerBuilder<byte[]> getConsumerConfiguration(PulsarClient client)
 
     @Override
     protected Boolean isAuthorized(String authRole, AuthenticationDataSource authenticationData) throws Exception {
-        return service.getAuthorizationService().canConsume(topic, authRole, authenticationData,
-                this.subscription);
+        try {
+            AuthenticationDataSubscription subscription = new AuthenticationDataSubscription(authenticationData,
+                    this.subscription);
+            return service.getAuthorizationService()
+                    .allowTopicOperationAsync(topic, TopicOperation.CONSUME, authRole, subscription)
+                    .get(service.getConfig().getMetadataStoreOperationTimeoutSeconds(), SECONDS);
+        } catch (InterruptedException e) {
```

Comment:
InterruptedException > `ExecutionException | InterruptedException | TimeoutException`

Reply:
I catch exception in the next part catch block. The difference is just in the error message.

Reply:
I just noticed the catch is wrong though. I'll update it.

Reply:
Actually, I copied the code from a number of places, I am going to merge and open another PR.

Reply:
PR to clean up the error: https://github.com/apache/pulsar/pull/20349